pagopa/io-app
# chore(IT Wallet): [SIW-999] Add category filter tabs to the new wallet screen

## Short description
This PR adds the filter tabs in the new wallet section screen

## List of changes proposed in this pull request
- Added `WalletCategoryFilterTabs` component
- Added reducer and actions to manage cards category filter
- Moved `renderWalletCardFn` to `utils` folder
- If only one card present, `WalletCardsContainer` now renders the card without category container
- Added unit tests

## How to test
With the `io-dev-api-server`, enable the **New wallet section** FF from the profile screen. 
Check that the cards and the tabs behaves correctly like in the preview below. In particular, check that with only one card you could not see the filter tabs.

## Preview

| One category | Multiple categories |
| --- | --- |
| <img src="https://github.com/pagopa/io-app/assets/6160324/2f592230-dfec-417d-8bfc-f4751e2e5218" width="250" /> | <video src="https://github.com/pagopa/io-app/assets/6160324/c2388df3-5b6a-4166-95ef-47d61cf3a4fa" width="250" />
